tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/BasicCasesTest.java

```diff
@@ -55,31 +55,68 @@ public void testJavaInterops() throws BallerinaTestException {
         clientLeecher.waitForText(20000);
     }
 
+    @Test()
+    public void testAnnotationAccess() throws BallerinaTestException {
+        String msg = "3 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"annotation-access"}, null,
+                new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(40000);
+    }
+
+    @Test()
+    public void testAnnotations() throws BallerinaTestException {
+        String msg = "15 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"annotations"}, null,
+                new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(40000);
+    }
+
     @Test(dependsOnMethods = "testAssertTrue")
-    public void testAllExceptAssertTrue() throws BallerinaTestException {
-        String msg1 = "15 passing";
-        String msg2 = "1 passing";
-        String msg3 = "87 passing";
-        String msg4 = "3 passing";
-        String msg5 = "2 passing";
-        String msg6 = "3 passing";
-        String msg7 = "8 passing";
-        LogLeecher clientLeecher1 = new LogLeecher(msg1);
-        LogLeecher clientLeecher2 = new LogLeecher(msg2);
-        LogLeecher clientLeecher3 = new LogLeecher(msg3);
-        LogLeecher clientLeecher4 = new LogLeecher(msg4);
-        LogLeecher clientLeecher5 = new LogLeecher(msg5);
-        LogLeecher clientLeecher6 = new LogLeecher(msg6);
-        LogLeecher clientLeecher7 = new LogLeecher(msg7);
-        balClient.runMain("test", new String[]{"--disable-groups", "p1", "--all", "--", "--user.name=waruna"}, null,
-                          new String[]{}, new LogLeecher[]{clientLeecher1, clientLeecher2, clientLeecher3,
-                        clientLeecher4, clientLeecher5, clientLeecher6, clientLeecher7}, projectPath);
-        clientLeecher1.waitForText(400000);
-        clientLeecher2.waitForText(400000);
-        clientLeecher3.waitForText(400000);
-        clientLeecher4.waitForText(400000);
-        clientLeecher5.waitForText(400000);
-        clientLeecher6.waitForText(400000);
-        clientLeecher7.waitForText(400000);
+    public void testAssertions() throws BallerinaTestException {
+        String msg = "87 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--disable-groups", "p1", "assertions", "--user.name=waruna"}, null,
+                new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(40000);
+    }
+
+    @Test()
+    public void testBeforeAfter() throws BallerinaTestException {
```

Comment:
Assertions are used in all tests. So the order of test execution should be as follows:
1. testAssertTrue
2. testAssertions (the assertions other than assertTrue)
3. rest of the tests 

Therefore, this and the rest of the tests should depend on `testAssertions`

Reply:
+1

Reply:
Addressed requested changes.